convert2rhel/pkghandler.py

```diff
@@ -608,6 +609,32 @@ def remove_non_rhel_kernels():
     return non_rhel_kernels
 
 
+def fix_default_kernel():
+    """
+        On systems converted from OL 6 and latter or CentOS Plus, DEFAULTKERNEL may contain kernel-uek/kernel-plus
+        after the conversion.
+
+        The solution handled by this function is to remove the  kernel-uek/kernel-plus from DEFAULTKERNEL setting the
+        correct kernel according with the rhel version.
+        """
+    if system_info.arch == "s390x":
+        # Applicable only on systems using GRUB2 (s390x uses zipl)
+        return
+
+    loggerinst = logging.getLogger(__name__)
+
+    loggerinst.info("Checking remaining Oracle Linux/CentOS Plus kernel and fix to RHEL kernel...")
+    kernel_sys_cfg = utils.get_file_content("/etc/sysconfig/kernel")
+
+    kernel_to_change = ["kernel-uek", "kernel-plus"]
+    # need to change to "kernel" in rhel6, 7 and "kernel-core" in rhel8
+    new_kernel_str = "DEFAULTKERNEL=" + ("kernel" if int(system_info.version) in [6, 7] else "kernel-core")
+
+    for kernel in kernel_to_change:
+        kernel_sys_cfg = kernel_sys_cfg.replace("DEFAULTKERNEL=" + kernel, new_kernel_str)
+    utils.store_content_to_file("/etc/sysconfig/kernel", kernel_sys_cfg)
```

Comment:
Ideally add a check if string exists, then make a change, then output to logger.

Only update the file if a change was made, and if a change was made you should output of successful action to logger after saving the file